internal/inventory/awsfetcher/fetcher_ec2_instance.go

```diff
@@ -78,6 +78,10 @@ func (e *ec2InstanceFetcher) Fetch(ctx context.Context, assetChannel chan<- inve
 			})
 		}
 
+		subnetIds := []string{}
+		if id := pointers.Deref(instance.SubnetId); id != "" {
```

Comment:
Another function that I usually implement on the `pointers.go` in my personal projects is `pointers.IsEmpty` where it checks for nullability or equals the zero value of that type. 

I like it. Feel free to implement or not :)